src/world.rs

```diff
@@ -0,0 +1,214 @@
+use std::{
+    fs, io::Read, path::{Path, PathBuf}
+};
+
+use regex::Regex;
+use serde::Deserialize;
+
+use crate::{Error, Map, ResourceCache, ResourceReader};
+
+/// A World is a collection of maps and their layout in the game world.
```

Comment:
I think this could be a bit more accurate: IMO it cannot be considered a collection of maps because the object does not actually contain any maps. It could be a collection of references to maps, though.